index.html

```diff
@@ -2554,42 +2556,52 @@ <h3>
             The user agent is free to display the icon in any context.
           </dd>
         </dl>
+        <p class="note">
+          If an icon contains multiple purposes, it can be used for any of
+          those purposes. If all of the stated purposes are not recognized, the
+          icon is totally ignored. For example, if an icon has purpose
```

Comment:
This is a bit contradictory with the SHOULD above. Maybe the above SHOULD needs to be MUST, otherwise this doesn't hold true.

Reply:
No, they are talking about slightly different things.

The SHOULD above:

> User agents SHOULD NOT use an icon other than for its stated purpose.

Is saying you shouldn't use a "maskable" icon (i.e., a valid purpose) as a normal icon. Since this is UI, we can't upgrade this to a MUST.

Whereas this non-normative note:

> If all of the stated purposes are not recognized, the icon is totally ignored.

Is saying that the user agent must not under any circumstances consider an icon whose purpose is not in {"any", "badge", "maskable"}. The corresponding normative text for this is the algorithm below, which converts icons without any recognised purpose into failure.

Reply:
Ok, but then the algorithm takes care of dropping the unknown purposes on the floor. Doesn't that mean the SHOULD can go away? The reasoning being that a conforming user agent would have dropped the icons with unknwn purposes, and will only be left with a list of known icons to display.

Reply:
Actually, never mind. You are right that it's about stated purpose.